analyses/independent-samples-release/util/independent-dna-samples.R

```diff
@@ -0,0 +1,67 @@
+# independent-dna-samples.R
+
+#' Generate a vector of unique samples
+#' 
+#' The samples from this function will be unique with respect to participants
+#' i.e. only no two samples will come from the same participant. The input list 
+#' should be pre-filtered by `experimental_strategy` and `sample_type`.  
+#' 
+#' 
+#' @param sample_df A data frame of samples, with columns corresponding to those
+#'   in `histologies-base.tsv`
+#' @param tumor_types Designates which types of tumors will be included. Options
+#'   are "primary" to include only primary tumors, "prefer_primary" to include
+#'   primary tumors when available, but fall back to other types, or "any" to
+#'   randomly select among all available specimens. As of v11, primary tumors
+#'   are defined as those designated "Initial CNS Tumor", "Diagnosis", or "Primary Tumor" in the
+#'   `tumor_descriptor` field.
+#' @param seed An optional random number seed. 
+#' 
+#' @return a data frame of Participant and Specimen IDs, each present only once.
+independent_samples <- function(sample_df, 
+                                tumor_types = c("primary", "prefer_primary", "any"), 
+                                seed){
+  tumor_types <- match.arg(tumor_types)
+  if(!missing(seed)){set.seed(seed)}
+  
+  primary_descs <- c("Initial CNS Tumor", "Diagnosis", "Primary Tumor")
+  
+  sample_df <- sample_df %>%
+    dplyr::mutate(match_id = paste(Kids_First_Participant_ID, sample_id, sep = "_"))
```

Comment:
```suggestion
    dplyr::mutate(match_id = paste(Kids_First_Participant_ID, sample_id, cohort, sep = "_"))
```
I recommend including `cohort` in the `match_id`  string in addition to the `sample_id`. Some cohorts like `TARGET` and `GMKF` have identical `Kids_First_Participant_ID` (`USI ID`) and `sample_id`. This will lead to samples for a patient being selected from one cohort and not the other. Unless `each cohort` independent samples are not needed in this module.

Reply:
Thanks- we do not want to duplicate patients here